troposphere/static/js/components/modals/instance/launch/components/Image.react.js

```diff
@@ -1,96 +1,126 @@
+    getInitialState: function(){
+        let image = this.props.image;
+        let versionList = null;
+        let active = false;
+        if (image) {
+            versionList = image.get('versions');
 
-      getInitialState: function(){
-          let image = this.props.image;
-          let versionList = null;
-          let active = false;
-            if (image) {
-                versionList = image.get('versions');
-
-                if (versionList.length > 0) {
-                    active = true;
-                }
-            }
-            return {
-                active
+            if (versionList.length > 0) {
+                active = true;
             }
-      },
-
-      propTypes: {
-        image: React.PropTypes.instanceOf(Backbone.Model).isRequired,
-        onClick: React.PropTypes.func
-      },
-
-      handleClick: function () {
-        if (this.state.active) {
-            this.props.onSelectImage(this.props.image);
         }
-      },
+        return {
+            active
+        }
+    },
 
-      renderTags: function () {
-        var tags = stores.TagStore.getAll();
-        var activeTags = stores.TagStore.getImageTags(this.props.image);
+    propTypes: {
+    image: React.PropTypes.instanceOf(Backbone.Model).isRequired,
+    onClick: React.PropTypes.func
+    },
 
-        return (
-          <Tags tags={tags} activeTags={activeTags} renderLinks={false}/>
-        );
-      },
+    handleClick: function () {
+    if (this.state.active) {
+        this.props.onSelectImage(this.props.image);
+        return;
+    }
+    this.setState({ showAlert: true });
+    },
 
-      render: function () {
-        var image = this.props.image,
-          type = stores.ProfileStore.get().get('icon_set'),
-          imageCreationDate = moment(image.get('start_date')).format("MMM D, YYYY hh:mm a"),
-          iconSize = 67,
-          icon;
-        let fullDescription = image.get('description');
-        let renderDescription = fullDescription.length < 150 ? fullDescription : (fullDescription.substring(0,150) + " ...");
+    renderTags: function () {
+    var tags = stores.TagStore.getAll();
+    var activeTags = stores.TagStore.getImageTags(this.props.image);
 
-        // always use the Gravatar icons
-        icon = (
-          <Gravatar hash={image.get('uuid_hash')} size={iconSize} type={type}/>
-        );
+    return (
+        <Tags tags={tags} activeTags={activeTags} renderLinks={false}/>
+    );
+    },
 
-        let inactiveClass = !this.state.active ? "disabled" : "";
+    render: function () {
+    var image = this.props.image,
+        type = stores.ProfileStore.get().get('icon_set'),
+        imageCreationDate = moment(image.get('start_date')).format("MMM D, YYYY hh:mm a"),
+        iconSize = 67,
+        icon;
+    let fullDescription = image.get('description');
+    let renderDescription = fullDescription.length < 150 ? fullDescription : (fullDescription.substring(0,150) + " ...");
```

Comment:
I'll do a good reformat. Been trying to discipline myself not to reformat while I work and do it in a separate commit. Just need to remember to actually do that commit.